scm_integrations.go

```diff
@@ -0,0 +1,108 @@
+package scalingo
+
+import (
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-scalingo/http"
+)
+
+type SCMType string
+
+// Type of SCM integrations
+const (
+	SCMGithubType           SCMType = "github"             // GitHub
+	SCMGithubEnterpriseType SCMType = "github-enterprise"  // GitHub Enterprise (private instance)
+	SCMGitlabType           SCMType = "gitlab"             // GitLab.com
+	SCMGitlabSelfHostedType SCMType = "gitlab-self-hosted" // GitLab self-hosted (private instance)
+)
+
+func (t SCMType) Str() string {
+	return string(t)
+}
+
+type SCMIntegrationsService interface {
+	SCMIntegrationsList() ([]SCMIntegration, error)
+	SCMIntegrationsCreate(scmType SCMType, url string, accessToken string) (*SCMIntegration, error)
+	SCMIntegrationsDestroy(id string) error
+}
+
+var _ SCMIntegrationsService = (*Client)(nil)
+
+type SCMIntegration struct {
+	ID          string  `json:"id,omitempty"`
```

Comment:
```suggestion
	ID          string  `json:"id"`
```

Are you sure about the `omitempty`? An ID seems mandatory to me

Reply:
The ID is automatically generated by the micro-service who manage it.

This struct is only used in the return of theses methods: list, show and create.

In create method, `scm_type` and `access_token` is mandatory to create an integration.